SourceX/controls/devices/kbcontroller.h

```diff
@@ -7,15 +7,25 @@
 
 #if HAS_KBCTRL == 1
 #include <SDL.h>
+#include "controls/controller.h"
 #include "controls/controller_buttons.h"
 
 namespace dvl {
 
-ControllerButton KbCtrlToControllerButton(const SDL_Event &event);
+class KeyboardController : public Controller {
+	static std::vector<KeyboardController> *const keyboardControllers_;
 
-bool IsKbCtrlButtonPressed(ControllerButton button);
+public:
+	static void Add(int device_index);
+	static KeyboardController *Get(const SDL_Event &event);
+
+	ControllerButton ToControllerButton(const SDL_Event &event) const;
 
-bool ProcessKbCtrlAxisMotion(const SDL_Event &event);
+	//bool ProcessKbCtrlAxisMotion(const SDL_Event &event);
+	bool ProcessAxisMotion(const SDL_Event &event);
+};
+
+bool IsKbCtrlButtonPressed(ControllerButton button);

```

Comment:
`const`?

Reply:
Where?

Reply:
There's no need for `const` on the argument -- it is passed by value (whether the method modifies it or not is an implementation detail)
There's no need for `const` on the return type -- it is returned by value (what the caller does with the result is the caller's business)

Reply:
I meant `const` as in `pure`, but I realize now this is not a method but a global function. Disregard.